clientconn_test.go

```diff
@@ -69,6 +69,22 @@ func TestTLSDialTimeout(t *testing.T) {
 	}
 }
 
+func TestTLSServerNameOverwrite(t *testing.T) {
+	overwriteServerName := "over.write.server.name"
+	creds, err := credentials.NewClientTLSFromFile(tlsDir+"ca.pem", overwriteServerName)
+	if err != nil {
+		t.Fatalf("Failed to create credentials %v", err)
+	}
+	conn, err := Dial("Non-Existent.Server:80", WithTransportCredentials(creds), WithTimeout(time.Millisecond))
+	if err != nil {
+		t.Fatalf("grpc.Dial(_, _) = _, %v, want _, <nil>", err)
+	}
+	if conn.authority != overwriteServerName {
+		t.Fatalf("conn.authority = %v, want %v", conn.authority, overwriteServerName)
+	}
+	conn.Close()
```

Comment:
move this after the error check and `defer` it to prevent leaking goroutines when this test fails